my_pyscf/mcscf/lasci_sync.py

```diff
@@ -1106,33 +1110,38 @@ def get_veff (self, dm1s_mo=None):
         moH = mo.conjugate ().T
         nmo = mo.shape[-1]
         dm1_mo = dm1s_mo.sum (0)
-        if getattr (self, 'bPpj', None) is None:
-            dm1_ao = np.dot (mo, np.dot (dm1_mo, moH))
-            veff_ao = np.squeeze (self.las.get_veff (dm1s=dm1_ao))
-            return np.dot (moH, np.dot (veff_ao, mo)) 
-        ncore, nocc, ncas = self.ncore, self.nocc, self.ncas
-        # vj
-        t0 = (lib.logger.process_clock (), lib.logger.perf_counter ())
-        veff_mo = np.zeros_like (dm1_mo)
-        dm1_rect = dm1_mo + dm1_mo.T
-        dm1_rect[ncore:nocc,ncore:nocc] /= 2
-        dm1_rect = dm1_rect[:,:nocc]
-        rho = np.tensordot (self.bPpj, dm1_rect, axes=2)
-        vj_pj = np.tensordot (rho, self.bPpj, axes=((0),(0)))
-        t1 = lib.logger.timer (self.las, 'vj_mo in microcycle', *t0)
-        dm_bj = dm1_mo[ncore:,:nocc]
-        vPpj = np.ascontiguousarray (self.las.cderi_ao2mo (mo, mo[:,ncore:]@dm_bj, compact=False))
-        # Don't ask my why this is faster than doing the two degrees of freedom separately...
-        t1 = lib.logger.timer (self.las, 'vk_mo vPpj in microcycle', *t1)
-
         if gpu:
-            naux = self.bPpj.shape[0]
-            vk_bj = np.zeros( (nmo-ncore, nocc) )
-            libgpu.libgpu_hessop_get_veff(gpu, naux, nmo, ncore, nocc, self.bPpj, vPpj, vk_bj)
-            t1 = lib.logger.timer (self.las, 'vk_mo (bb|jj) in microcycle', *t1)   
-            t1 = lib.logger.timer (self.las, 'vk_mo (bi|aj) in microcycle', *t1);
-
-        else: 
+            dm1_ao=np.dot(mo,np.dot(dm1_mo,moH))
+            veff_ao=np.squeeze(self.las.get_veff(dm1s=dm1_ao))
+            return np.dot(moH,np.dot(veff_ao,mo))
+        else:
+            if getattr (self, 'bPpj', None) is None:
+                dm1_ao = np.dot (mo, np.dot (dm1_mo, moH))
```

Comment:
1119-1121 is the same as 1114-1116. Why not put it in the same conditional block? Why indent everything? ("Return" ends the function execution; a "return" at the end of an "if" block means you don't need the "else" block)

Reply:
OK I made a cleaner block on my fork. Do I just open another pull request?

Reply:
This PR will track any update you push to valay1:valay_gpu as long as it stays open.